### DISCUSSION TITLE
5.0.0 Release

### DISCUSSION THREAD

**Discussion by Androz2091** [Androz2091/discord-player#663] [General]
**[v4.x to v5.x migration guide](https://discord-player.js.org/docs/main/master/migrating/migrating)**

This version of discord-player has been rewritten using @discordjs/voice, for better performance. It also now supports Discord.js v13 and Slash Commands. An important part of the core of discord-player has been rebuilt and this leads to many breaking changes. This is something we can't escape if we want to add major improvements to the library.

Thank you for using discord-player, we look forward to your feedback! If you have any questions, feel free to join our [Discord](https://androz2091.fr/discord) server.

## Breaking ⚠️

* Discord.js v13 is required.
* Node.js v16 is required.
* All the methods are no longer based on message
* `ytdlDownloadOptions` has been renamed to `ytdlOptions`.
* Most of the options from Player class has been moved to Queue.
* All the event parameters have been changed.
* Provides cleaner API
* ESM support
* Each player methods are now handled by individual queues
* leaveOnEndCooldown option has been removed
* leaveOnStopCooldown option has been removed

## Added

* New metadata option to all the Queue instances
* New initialVolume option to all the Queue instances
* New bufferingTimeout option for Player
* New connectionTimeout option for voice connections
* New audio filters
* Support for Slash Commands (/)
* Support for Stage channels
* New Playlist structure

## Improved

* Better error handling
* Better tracks search
* Better voice state handling
* Better queue management

<hr /><em>This discussion was created from the release <a href='https://github.com/Androz2091/discord-player/releases/tag/v5.0.0'>5.0.0</a>.</em>

**Comment by twlite** [comment]
👀

**Comment by DawnOfDedSec** [comment]
@Andromeda#6633whenever I try to add a track during `queue.playing` is true , then music stops but neither it is paused or stopped , its like it is frozen , btw I do it like - `queue.addTrack(track)`

**Comment by ConnorDoesDev** [comment]
Pog! I have a question though, why did you remove the `leaveOnEndCooldown` and the `leaveOnStopCooldown` options?
